src/core/src/handle.rs

```diff
@@ -32,15 +32,13 @@ impl<R: Resources> ops::Deref for RawBuffer<R> {
 }
 
 /// Type-safe buffer handle
-#[derive(Clone, Debug)]
+#[derive(Clone, Debug, Eq)]
 pub struct Buffer<R: Resources, T>(RawBuffer<R>, PhantomData<T>);
 
 impl<R: Resources, T> cmp::PartialEq for Buffer<R, T> {
     fn eq(&self, other: &Self) -> bool { self.0.eq(&other.0) }
 }
 
-impl<R: Resources, T> cmp::Eq for Buffer<R, T> {}
```

Comment:
we shouldn't be removing this line. The difference with `derive` is that we don't require `T: Eq` here, which is essential